File: hw/fuentes/U3/HW17Maze/NewMazeGenerator/src/main/java/ec/edu/espe/mazegenerator/model/Maze.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.mazegenerator.model;

/**
 *
 * @author jorge
 */
public class Maze {
    public static int N;
    public static int M;
    private Room[][] rooms;
    private Room entrance;
    private Room exit;

    public Maze(int N, int M) {
        Maze.N = N;
        Maze.M = M;
        this.rooms = new Room[N][M];
        initializeRooms();
    }

    private void initializeRooms() {
        for (int i = 0; i < N; i++) {
            for (int j = 0; j < M; j++) {
                rooms[i][j] = new Room(i, j);
            }
        }
    }

    public Room getRoom(int x, int y) {
        if (x >= 0 && x < N && y >= 0 && y < M) {
            return rooms[x][y];
        }
        return null;
    }

    public Room[][] getRooms() {
        return rooms;
    }

    public void setEntrance(Room entrance) {
        this.entrance = entrance;
    }

    public Room getEntrance() {
        return entrance;
    }

    public void setExit(Room exit) {
        this.exit = exit;
    }

    public Room getExit() {
        return exit;
    }

    public String toAscii() {
        
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < N; i++) {
            for (int j = 0; j < M; j++) {
                sb.append("[ ]");
            }
            sb.append("\n");
        }
        return sb.toString();
    }
    
}

File: hw/fuentes/U3/HW17Maze/MazeAppPY/ec/edu/espe/Maze/model/Maze.py
from model.Room import Room

class Maze:
    N = 0
    M = 0

    def __init__(self, n, m):
        Maze.N = n
        Maze.M = m
        self.rooms = [[Room(i, j) for j in range(m)] for i in range(n)]
        self.entrance = None
        self.exit = None

    def get_room(self, x, y):
        if 0 <= x < Maze.N and 0 <= y < Maze.M:
            return self.rooms[x][y]
        return None

    def get_rooms(self):
        return self.rooms

    def set_entrance(self, room):
        self.entrance = room

    def get_entrance(self):
        return self.entrance

    def set_exit(self, room):
        self.exit = room

    def get_exit(self):
        return self.exit

    def to_ascii(self):
        result = ""
        for i in range(Maze.N):
            for j in range(Maze.M):
                result += "[ ]"
            result += "\n"
        return result
